apps/frontend/src/app/PageTeam/CharacterDisplay/Tabs/TabOptimize/index.tsx

```diff
@@ -194,10 +195,15 @@ export default function TabBuild() {
       levelHigh,
       allowLocationsState,
       useExcludedArts,
+      useLoadoutExclusion,
+      loadoutExclusion,
     } = deferredArtsDirty && deferredBuildSetting
 
+    console.log('Run Filter')
     return database.arts.values.filter((art) => {
       if (!useExcludedArts && artExclusion.includes(art.id)) return false
+      if (!useLoadoutExclusion && loadoutExclusion.includes(art.id))
```

Comment:
This should be where you apply the logic in the `useEffect` above.

Reply:
Can you explain more what you mean!

Reply:
sure, the code here should look like:
```
const artsOnTeammates = teammates.flatmap(tm=>db.getLoadoutArtifacts.id)


return database.arts.values.filter(art=>{
...
if(!useTeammatesBuild && artsOnTeammates.includes(art.id)) return false
...
})

```